BGL/test/BGL/test_Collapse_edge.cpp

```diff
@@ -53,14 +53,19 @@ collapse_edge_test()
                                           0.5,0,
                                           test_mesh);
     halfedge_descriptor en = next(e, test_mesh);
-    halfedge_descriptor eno = opposite(nesxt(e, test_mesh), test_mesh);
+    halfedge_descriptor eno = opposite(en, test_mesh);
+    halfedge_descriptor eno_prime = opposite(next(opposite(e, test_mesh), test_mesh), test_mesh);
     vertex_descriptor v1 = target(edge(e, test_mesh), test_mesh);
-    char found =0;
+    assert(CGAL::Euler::collapse_edge(edge(e, test_mesh), test_mesh) == v1);
+    char found = 0;
     BOOST_FOREACH(halfedge_descriptor it, CGAL::halfedges_around_target(v1,test_mesh))
     {
-
+      if(it == eno
+         || it == eno_prime){
+        ++found;
+      }
     }
-    assert(CGAL::Euler::collapse_edge(edge(e, test_mesh), test_mesh) == v1);
+    assert(found == 2);
```

Comment:
This is only true if e and its opposite or not boundary edges, otherwise it is 1.

Reply:
Sorry, I though you were testing next of e and next of opposite(e) (which you should be testing).